parquet-column/src/main/java/org/apache/parquet/io/api/Binary.java

```diff
@@ -505,31 +488,12 @@ boolean equals(ByteBuffer otherBytes, int otherOffset, int otherLength) {
 
     @Override
     public int compareTo(Binary other) {
-      if (value.hasArray()) {
-        return other.compareTo(value.array(), value.arrayOffset() + offset, length);
-      } else {
-        return other.compareTo(value, offset, length);
-      }
-    }
-
-    @Override
-    int compareTo(byte[] other, int otherOffset, int otherLength) {
-      if (value.hasArray()) {
-        return Binary.compareTwoByteArrays(value.array(), value.arrayOffset() + offset, length,
-            other, otherOffset, otherLength);
-      } {
-        return Binary.compareByteBufferToByteArray(value, offset, length, other, otherOffset, otherLength);
-      }
-    }
-
-    @Override
-    int compareTo(ByteBuffer bytes, int otherOffset, int otherLength) {
-      return Binary.compareTwoByteBuffers(value, offset, length, bytes, otherOffset, otherLength);
+      return PrimitiveComparator.LEXICOGRAPHICAL_BINARY_COMPARATOR.compare(this, other);
     }
 
     @Override
     public ByteBuffer toByteBuffer() {
-      ByteBuffer ret = value.slice();
+      ByteBuffer ret = value.duplicate();
```

Comment:
Why was this change necessary?

Reply:
`slice()` returns a `ByteBuffer` that refers to a part of the original one (mark and position are reset, limit will be the remaining of the original as well as capacity etc.). We need the "original" `ByteBuffer` for comparing just as the other implementations provide it.
(That's why you've done the same modification [here](https://github.com/apache/parquet-mr/pull/390/files#diff-a7b135229f233fb724e70eae501f97c3R532) ;) )

Reply:
Right, but why was this necessary in this PR?

Reply:
Yes, it failed the tests of the binary comparators.